src/components/shared/RecentJobsList.vue

```diff
@@ -1,29 +1,47 @@
 <script>
 import { mapState, mapActions } from 'vuex';
 import JobListing from './JobListing';
+import JobPeriodSelector from './JobPeriodSelector';
+import { PERIODS } from '../../store/constants';
 
 export default {
   components: {
     JobListing,
+    JobPeriodSelector,
   },
   data() {
     return {
       isLoading: false,
     };
   },
   computed: {
-    ...mapState(['recentPosts']),
+    ...mapState(['recentPosts', 'activePeriod']),
   },
   methods: {
-    ...mapActions(['fetchRecentPosts']),
+    ...mapActions(['fetchRecentPosts', 'setPeriod']),
+    fetch() {
+      this.isLoading = true;
+
+      this.fetchRecentPosts()
+        .then(() => {
+          this.isLoading = false;
+        });
+    },
+    handlerSetPeriod() {
+      const period = PERIODS.find(p => p.slug === this.$route.params.slug);
+      if (period) {
+        this.setPeriod(period.type);
+      }
+      this.fetch();
```

Comment:
This line should also be in the if block above.

Reply:
Actually no, it shouldn't as you mentioned in Slack. We call this method in the created hook so it should do the fetch in any case.